### ISSUE TITLE
Questions about the implementation of Grad-CAM++: The second and third order derivatives

### ISSUE THREAD

**Issue by hxngu** [jacobgil/pytorch-grad-cam#557]
#541 When I try to learn how to get hessian matrix I find the implementation of Grad-CAM++ just used `grads_power_2 = grads**2` and `grads_power_3 = grads_power_2 * grads` to get the second and third order derivatives. Is that correct? The formula in original paper is listed below.

<img width="389" alt="Image" src="https://github.com/user-attachments/assets/c3a3db66-8db2-45b5-99ef-090536bcad94" />

**Comment by caihuaiguang** [comment]
GradCAM++ approximates second and third-order derivatives using squared and cubed gradients, which holds only when logits are passed through an exponential function (i.e., output = exp(logits) ). 

Computing the exact Hessian matrix is both time-consuming and space-consuming. However, obtaining the Hessian-vector product requires only a single backward pass, as demonstrated in [ShapleyCAM](https://arxiv.org/abs/2501.06261), which is also merged to this repository.

**Comment by caihuaiguang** [comment]
I believe this issue arises because the default setting is `targets = [ClassifierOutputTarget(281)]` (i.e., pre-softmax utility function). To avoid get None, you can use ` targets = [ClassifierOutputReST(281)]` instead.

This is because for most CNN architectures, the Hessian matrix computed using pre-softmax is zero. This is due to the fact that operations like ReLU and convolution are linear. The Hessian only becomes non-zero when a nonlinear activation function is involved, such as EfficientNet.

For more details,  please check this response:
[https://github.com/jacobgil/pytorch-grad-cam/pull/550#issuecomment-2600889939](https://github.com/jacobgil/pytorch-grad-cam/pull/550#issuecomment-2600889939)

**Comment by hxngu** [comment]
Thanks a lot for your quick reply.
